src/Pmi/Service/PatientStatusService.php

```diff
@@ -28,29 +29,75 @@ public function sendPatientStatusToRdr()
                    psh.status,
                    psh.comments,
                    psh.created_ts as authored,
+                   psh.import_id,
                    ps.participant_id,
                    ps.organization,
                    ps.awardee,
                    u.email as user_email
             FROM patient_status_history psh
             INNER JOIN patient_status ps ON ps.id = psh.patient_status_id
             LEFT JOIN users u ON psh.user_id = u.id
-            WHERE psh.rdr_ts is null
+            WHERE psh.rdr_ts is null and psh.rdr_status = 0
             ORDER BY psh.id ASC
             limit {$limit}
         ";
         $patientStatusHistories = $this->em->fetchAll($query, []);
         $patientStatusObj = new PatientStatus($this->app);
+        $importIds = [];
         foreach ($patientStatusHistories as $patientStatusHistory) {
+            if (!in_array($patientStatusHistory['import_id'], $importIds)) {
+                $importIds[] = $patientStatusHistory['import_id'];
+            }
             $patientStatusObj->loadDataFromDb($patientStatusHistory);
             if ($patientStatusObj->sendToRdr()) {
-                $this->em->getRepository('patient_status_history')->update($patientStatusHistory['id'], ['rdr_ts' => new \DateTime()]);
+                // Set rdr_status = 1 for success
+                $this->em->getRepository('patient_status_history')->update($patientStatusHistory['id'], ['rdr_ts' => new \DateTime(), 'rdr_status' => 1]);
                 $this->app->log(Log::PATIENT_STATUS_HISTORY_EDIT, [
                     'id' => $patientStatusHistory['id']
                 ]);
             } else {
                 $this->app['logger']->error("#{$patientStatusHistory['id']} failed sending to RDR: " . $this->rdr->getLastError());
+                // RDR status
+                // 2 = RDR 400 (Invalid participant id)
+                // 3 = RDR 500 (Invalid patient status)
+                // 4 = Other RDR errors
+                $rdrStatus = 4;
+                if ($this->rdr->getLastErrorCode() === 400) {
+                    $rdrStatus = 2;
+                } elseif ($this->rdr->getLastErrorCode() === 500) {
+                    $rdrStatus = 3;
+                }
```

Comment:
A few comments on the status:

1. Can use class constants (maybe in the `PatientStatusHistory` entity class) to reference these integers? That way we don't have to use these arbitrary status values throughout the code and it is more immediately obvious what the statuses mean.
2. We don't know for sure that a 500 error is always going to be because of an invalid patient status. So the comment here (or class constant if (1) is implemented) should reflect that.